File: my-folder/0232-implement-queue-using-stacks/solution.js
var MyQueue = function() {
    this.s1 = [];
    this.s2 = [];
};

/** 
 * @param {number} x
 * @return {void}
 */
MyQueue.prototype.push = function(x) {
    this.s1.push(x);
};

/**
 * @return {number}
 */
MyQueue.prototype.pop = function() {
    if (this.s1.length === 0) return null;

    while (this.s1.length > 0) {
        const x = this.s1.pop();
        this.s2.push(x);
    }

    const res = this.s2.pop();

    while (this.s2.length > 0) {
        const x = this.s2.pop();
        this.s1.push(x);
    }

    return res;
};

/**
 * @return {number}
 */
MyQueue.prototype.peek = function() {
    if (this.s1.length === 0) return null;

    while (this.s1.length > 0) {
        const x = this.s1.pop();
        this.s2.push(x);
    }

    const res = this.s2.pop();
    this.s1.push(res);

    while (this.s2.length > 0) {
        const x = this.s2.pop();
        this.s1.push(x);
    }

    return res;
};

/**
 * @return {boolean}
 */
MyQueue.prototype.empty = function() {
    return this.s1.length === 0;
};

/** 
 * Your MyQueue object will be instantiated and called as such:
 * var obj = new MyQueue()
 * obj.push(x)
 * var param_2 = obj.pop()
 * var param_3 = obj.peek()
 * var param_4 = obj.empty()
 */

File: my-folder/0232-implement-queue-using-stacks/solution.py
class MyQueue:

    def __init__(self):
        self.primary_stack = []
        self.secondary_stack = []
        self.first = None

    def push(self, x: int) -> None:
        self.first = x if not self.first else self.first
        self.primary_stack.append(x)

    def pop(self) -> int:
        while(self.primary_stack):
            self.secondary_stack.append(self.primary_stack.pop())
        
        first = self.secondary_stack.pop()
        self.first = self.secondary_stack.pop() if self.secondary_stack else None
        
        if self.first:
            self.primary_stack.append(self.first)

            while(self.secondary_stack):
                self.primary_stack.append(self.secondary_stack.pop())

        return first

    def peek(self) -> int:
        return self.first

    def empty(self) -> bool:
        return len(self.primary_stack) == 0
        


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()
